settings/rest-api.php

```diff
@@ -0,0 +1,125 @@
+
+/**
+ * Register/Output some REST-API calls to be pre-loaded.
+ *
+ * This prevents the browser having to make the HTTP call before the react UI can be ready.
+ * This duplicates block_editor_rest_api_preload() as there is no generic function for performing this preloading.
+ * WARNING: This will output Javascript immediately if called during the page load if the wp-api-fetch script has already been output.
+ *
+ * @param array $preload_paths The REST API paths to be pre-loaded, must include prefixed slash.
+ * @return void
+ */
+function preload_api_requests( array $preload_paths ) : void {
+	global $post, $wp_scripts, $wp_styles;
+
+	/*
+	 * Ensure the global $post, $wp_scripts, and $wp_styles remain the same after
+	 * API data is preloaded.
+	 * Because API preloading can call the_content and other filters, plugins
+	 * can unexpectedly modify the global $post or enqueue assets which are not
+	 * intended for the block editor.
+	 */
+	$backup_global_post = ! empty( $post ) ? clone $post : $post;
+	$backup_wp_scripts  = ! empty( $wp_scripts ) ? clone $wp_scripts : $wp_scripts;
+	$backup_wp_styles   = ! empty( $wp_styles ) ? clone $wp_styles : $wp_styles;
+
+	$preload_data = array_reduce(
+		$preload_paths,
+		'rest_preload_api_request',
+		[]
+	);
+
+	// Restore the global $post, $wp_scripts, and $wp_styles as they were before API preloading.
+	$post       = $backup_global_post;
+	$wp_scripts = $backup_wp_scripts;
+	$wp_styles  = $backup_wp_styles;
+
+	$preload_js = sprintf(
+		'wp.apiFetch.use( wp.apiFetch.createPreloadingMiddleware( %s ) );',
+		wp_json_encode( $preload_data )
+	);
+
+	// The script may have already been printed. In that case, add the pre-loading inline now.
+	if ( wp_script_is( 'wp-api-fetch', 'done' ) ) {
+		printf(
+			'<script>%s</script>',
+			$preload_js
+		);
+	} else {
+		wp_enqueue_script( 'wp-api-fetch' );
+		wp_add_inline_script(
+			'wp-api-fetch',
+			$preload_js,
+			'after'
+		);
+	}
+}
+
+/**
+ * Register any user meta that needs to be exposed.
+ */
+function register_user_fields(): void {
+	// Expose the `_new_email` user meta through the rest api as a custom user field.
+	// This is for "The user has a pending email change"
+	register_rest_field(
+		'user',
+		'pending_email',
+		[
+			'get_callback' => function( $user ) {
+				return get_user_meta( $user['id'], '_new_email', true )['newemail'] ?? false;
+			},
+			'update_callback' => function( $value, $user ) {
+				if ( '' === $value ) {
+					delete_user_meta( $user->ID, '_new_email' );
+					return true;
+				}
+			},
+			'schema' => [
+				'type'    => 'string',
+				'context' => [ 'edit' ],
+			]
+		]
+	);
+
+}
+
+/**
+ * Implement the "Require email confirmation" functionality for the REST API.
+ *
+ * TODO: This is a core bug. This should be handled by core.
+ * TODO: This should be moved to a WordPress.org mu-plugin.
+ * TODO: This generates urls to /support/wp-admin/profile.php?newuseremail=%s
```

Comment:
```suggestion
 * TODO: This generates urls to /support/wp-admin/profile.php?newuseremail=%s, but we want `get_edit_account_url()` instead
```

I missed that it was going to `wp-admin` at first glance, so explicitly saying what we want instead makes it more obvious

Reply:
TBH I don't want it going to that URL either really.. I think it should go to `login.wordpress.org/wp-login.php?confirmaction=blahblah` instead, and we could then have that redirect to a page of our choosing afterwards.

I need to make a Core PR to move the email confirmation code there instead of us adding special handling.

Reply:
👍🏻